File: app/src/main/java/org/lenchan139/lightbrowser/UrlFromExternalActivity.java
package org.lenchan139.lightbrowser;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

public class UrlFromExternalActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //setContentView(R.layout.activity_url_from_external);
        String inURL;
        if(getIntent().getData()!= null){
            Log.v(getString(R.string.KEY_INURL_INTENT),getIntent().getData().toString());
            inURL = getIntent().getData().toString();
            Intent intent = new Intent(this,MainActivity.class);
            intent.putExtra("InURL",inURL);
            startActivity(intent);

        }
        finish();

    }
}

File: app/src/main/java/org/lenchan139/lightbrowser/UrlFromExternalActivity.kt
package org.lenchan139.lightbrowser

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class UrlFromExternalActivity : AppCompatActivity() {

    override fun onPause() {
        super.onPause()
        overridePendingTransition(0, 0)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_url_from_external);
        var inURL: String?
        if (intent.data != null) {
            Log.v(getString(R.string.KEY_INURL_INTENT), intent.data.toString())
            inURL = intent.data.toString()
            val intent = Intent(this, MainActivity::class.java)
            intent.putExtra("InURL", inURL)
            startActivity(intent)
            inURL = null

        }
        finish()

    }
}
